### DISCUSSION TITLE
Storage place for macros

### DISCUSSION THREAD

**Discussion by databayes** [TabularEditor/TabularEditor3#1303] [Ideas]
### Is your feature request related to a problem? Please describe

Spent yesterday inside Citrix environment setting up macros for TE3.  This morning all have been wiped out, can't store in the Local AppData environment.  Is it possible to set storage anywhere else?  NBs it's taken 12 months to be allowed to have TE and wasn't able to download the portable option either.

### Describe the solution you'd like

Is it possible to set storage anywhere else?  The ideal would be to have a storage place that I could keep all macros in one place that I can point to regardless of which environment I work in.  Appreciate that this may be impossible for security reasons.

### Goals

1.  Ability to chose location of macro storage
2.

**Comment by edhans** [comment]
I would _love_ for these to be stored in the cloud (a royal hassle I suspect) or at least within a user folder in My Documents that syncs between machines via ODfB.

**Comment by otykier** [comment]
Good suggestion - we're considering how best to implement this. @databayes would it be sufficient to be able to specify another local storage location, such as the Documents folder as suggested by @edhans, or would Documents also be wiped in your specific case (Citrix)?

**Comment by databayes** [comment]
Thanks for the quick response, that would be great to be able to save in Documents, the best possible would be to have a cloud site that I could access from anywhere for any of my models but appreciate that's a big ask,
Sue 🙂



Book time with ***@***.***/meetingtype/mZSgy-721ECGLLNLD3JNdA2?anonymous&ep=mcard>


Sue Bayes, Director, www.databayes.co.uk<http://www.databayes.co.uk/>  T: 07792 547428  LinkedIn: https://www.linkedin.com/in/suebayes/

Microsoft MVP Data Platform

Microsoft Certified: Fabric Analytics Engineer Associate

Microsoft Certified: Azure Enterprise Data Analyst Associate

Microsoft Certified: Data Analyst Associate




***@***.***?anonymous&ep=signature>             Book time to meet with ***@***.***?anonymous&ep=signature>
________________________________

**Comment by Eivind4** [comment]
Maybe it should be a separate idea, but with the ability for selection for another storage location (also locally):  

Could it be possible to switch or append MacroActions, similar to functionality for Best Practice Analyzer? Like the idea to develop a common set of macros for our organization. But I assume it could be a challenge/conflict if I want to develop and test my own without using the organization MacroAction. As an organization there could be a way to split into some specific set of macros, such as for a vertical (Sales etc). Not sure as of today, if we are to organize with common MacroAction file, how I can use some that I like to use myself without having to get it available through the organizational standard. Assume it then could be overwritten when organizational standard is updated, such that I will loose The Snek game and Quiz, that would may not make it to the organizational MacroAction ;-)

**Reply by otykier** [reply]
Please see the discussion [here](https://github.com/TabularEditor/TabularEditor3/issues/106#issuecomment-876802187). The bottom line is that from a security POV, we cannot allow the application to load macros from an "untrusted" location. Sorry!

**Reply by Eivind4** [reply]
Did not find the discussion you sent when searching. Your link gave additional input on how this can be managed in a good way. Thank you for the good and quick reply!

**Reply by KornAlexander** [reply]
totally agree, with this. Serialize the Macros instead of the big MacroAction.json file is key. 

If this is not implemented I would need to improve my current powershell doing the merging. 
If this is available improving the official TE 3 repo actually would makes sense.
